src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -30,6 +31,8 @@ public partial class CSharpCompilation
         /// </summary>
         private Symbol[] _lazyWellKnownTypeMembers;
 
+        internal bool NeedsGeneratedIsReadOnlyAttribute { get; private set; }
```

Comment:
>internal bool NeedsGeneratedIsReadOnlyAttribute { get; private set; } [](start = 8, length = 69)

It would be good to have a comment when one should expect this property to have actual value. We should probably have a way to "freeze" the value and a way to assert an attempt to set the value it after it was frozen and an attempt to read the value before it is frozen.